src/Microsoft.ComponentDetection.Orchestrator/Experiments/ExperimentService.cs

```diff
@@ -59,21 +61,46 @@ public void RecordDetectorRun(IComponentDetector detector, IEnumerable<DetectedC
         }
     }
 
+    private void FilterExperiments(IComponentDetector detector, int count) =>
+        this.experiments.RemoveAll(experiment =>
+        {
+            var shouldRemove = !experiment.Config.ShouldRecord(detector, count);
+
+            if (shouldRemove)
+            {
+                this.logger.LogDebug(
+                    "Removing {Experiment} from active experiments because {Id} has {Count} components",
```

Comment:
`has {Count} components` is true right now, but maybe not in general? IMO it's fine for now.

Reply:
Oop. This can be confusing if we filter experiments based on something other than components detected.